System/PathFinder.h

```diff
@@ -116,8 +115,17 @@ namespace RTE {
 		/// Recalculates the costs between all the nodes touching a deque of specific rectangular areas (which will be wrapped). Also resets the pather itself, if necessary.
 		/// </summary>
 		/// <param name="boxList">The deque of Boxes representing the updated areas.</param>
-		/// <returns>Whether any costs were updated.</returns>
-		bool RecalculateAreaCosts(const std::deque<Box> &boxList);
+		/// <param name="nodeUpdateLimit">The maximum number of nodes we'll try to update this frame. True node update count can be higher if we received a big box, as we always do at least 1 box.</param>
+		/// <returns>The set of nodes that were updated.</returns>
+		std::unordered_set<int> RecalculateAreaCosts(std::deque<Box> &boxList, int nodeUpdateLimit);
```

Comment:
Given how similar this is to RecalculateAllCosts, I wonder if you could make RecalculateAllCosts call this (with some flag to force consider nodes as changed), to avoid some code duplication.
Maybe that's kinda messy, but given how particular this stuff is (namely, the code to update up and left materials), it'd be good to have one source of truth for it imo.

Reply:
See what I mentioned about about constructing a big list of nodes. But valid point on a single-source-of-truth. I wouldn't had the funciton just take a pair of iterators (which means it's work on unordered_map and std::vector both), but as mentioned above, we require that intermediate copy into a vector... so unfortunately that doesn't work.

Reply:
Given the limitation you have where you can't use unordered_set in UpdateNodeList, I'd say it's worth changing RecalculateAreaCosts to not return an unordered_set at all. Instead, fill your unordered_set and copy it right away into a vector or whatever.

The unordered_set is only used with UpdateNodeList, which needs to copy everything into another DS anyway, and RecalculateAreaCosts even calls UpdateNodeList internally. If you do the copy once in RecalculateAreaCosts, you can avoid recopying it multiple times elsewhere, and you can make RecalculateAllCosts use UpdateNodeList under the hood.